doc/relevancy.md

```diff
@@ -0,0 +1,442 @@
+subsequent views to count less.  A typical modifier uses a log function.
+The available modifiers are used within the field_value_factor and are `none`
+(the default), `log`, `log1p`, `log2p`, `ln`, `ln1p`, `ln2p`, `square`, `sqrt`,
+and `reciprocal`.
+Currently we use a script_score instead which gives us more control.
+
+log1p example: `new_score = old_score * log(1 + number_of_votes)`
+
+Currently it is "doc['popularity'].value + #{POPULARITY_OFFSET}" but maybe it
+would be worth studying the inbuilt DSL modifiers to see how they could help us
+test out boosting with numbers/functions that are less made up.
+
+**Other ideas**
+
+Ways to change how original popularity score is set before it is boosted?
+
+The time period it is over could possibly be changed? Not sure how to pick a
+timeframe.
+
+
+## Query DSL usage
+
+Here is a map of what goes into our elasticsearch query: https://www.draw.io/#G0B3LRehFrM0QuaVRxV0hiZzhTbG8
+
+### The way we handle longer search terms is broken
+
+There are two ways the query takes into account phrases in search queries.
+[Phrase queries](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-match-query-phrase.html) and [shingles](https://www.elastic.co/blog/searching-with-shingles).
+
+We're not using the _slop_ parameter, so the phrase component will only work
+where the query uses the exact same phrasing. I'm not sure this is all that
+useful for us, except maybe for names of services/forms/departments.
+
+The `slop` parameter tells the `match_phrase` query how far apart terms are
+allowed to be while still considering the document a match. By _how far
+apart_ we mean _how many times do you need to move a term in order to make
+the query and document match?_
+
+Although all words need to be present in phrase matching, even when using
+slop, the words don’t necessarily need to be in the same sequence in order
+to match. With a high enough slop value, words can be arranged in any order.
+
+https://www.elastic.co/guide/en/elasticsearch/guide/current/slop.html
+
+Shingles are more lenient since we're only looking for pairs of words instead
+of requiring the whole phrase to match. **But our implementation is broken.**
+This is one of the things that was fixed in the new search weighting.
+
+See Richard's commit message: https://github.com/alphagov/search-api/commit/6ece793190932fee31be5437c4c56773dc358ff6
+
+>  The existing weighting only applies this analysis at search time, so the
+>  pairs it looks for never exist in the index. This commit instead
+>  creates a new pseudo-subfield for searchable text fields which is
+>  indexed and searched with shingles applied. The weights from searches
+>  across this field are used to boost scores of documents.
+
+### There is no boosting of individual field matches
+
+We treat a match in title, description, acronym and indexable content the same,
+but these are completely different things.
```

Comment:
The new text similarity algorithm should basically do this for us, probably worth A/B testing removing this field boosting.